bin/android-functions.sh

```diff
@@ -95,6 +95,12 @@ function test_javac_version() {
   local target_version=$2
   local jc_v
 
+  # Setting JAVA_HOME to /usr causes sdkmanager to hang
+  if [[ "$javac" == "/usr/bin/javac" ]] && [[ "$(host_os)" == "Darwin" ]]; then
```

Comment:
is `/usr/bin/javac` the real path? what does `realpath -f /usr/bin/javac` say?

Reply:
Are you thinking of `readlink`?

```bash
% readlink -f /usr/bin/javac
/usr/bin/javac
```

Reply:
I think it's there to help people run `java`/`javac` when they aren't in `$PATH`. If you have `$JAVA_HOME` set, it runs that version for you.

Reply:
No I meant realpath, but readlink will do it too.

I am curious why this doesn't work:

```sh
java -XshowSettings:properties -version 2>&1 > /dev/null | grep 'java.home' 
```

Reply:
I haven't tried that, how should it be used?

Reply:
Thinking about this more, the reason it hangs is probably because `/usr/bin/java` goes into an infinite loop of trying to run itself.
`/usr/bin/java` runs the java in `$JAVA_HOME/bin`.
If `JAVA_HOME` is `usr`, then `/usr/bin/java` is going to run itself.
We just need to avoid using `/usr/bin/java`,.
If `$JAVA_HOME` is set, we don't need it.
We don't need `java` in `$PATH`, we just need to test that `$JAVA_HOME` is valid.

Reply:
> I haven't tried that, how should it be used?

I think just like:

```sh
JAVA_HOME="$(java -XshowSettings:properties -version 2>&1 > /dev/null | grep 'java.home')"
```

Reply:
That's only going to work if there's already a `JAVA_HOME` set, and it has to be a valid version for it to be of any use.